src/app/collection/pages/collection-ncyte/components/stats/stats.component.ts

```diff
@@ -131,8 +145,30 @@ export class StatsComponent implements OnInit {
   handleChartNotHovered() {
     CHART_HOVERED = false;
   }
+
+   async getStatsForWITHCyber() {
+    if (this.collectionName !== 'withcyber') {
+return;
+}
+    if (!this.tagId) {
+return;
 }
 
+    const tagMetrics = await this.metricService.getTagMetrics(this.tagId);
+    this.objReleased = tagMetrics.releasedLearningObjects;
+    this.objDownload = tagMetrics.downloads;
+    this.authorCollection = tagMetrics.authors.length;
+  }
+
+  async getWITHCyberTagId() {
+      const url =  TAGS_ROUTES.GET_ALL_TAGS({ text: 'WITHCyber' });
+      const res = await fetch(url, { method: 'GET' });
+      const data: TagsResponse = await res.json();
+      const tag =  data.tags?.[0]?._id ?? null;
+      console.log('Tag ID: ', tag);
+      return tag;
+    }
+}
```

Comment:
I think we can clean up this function a bit, instead of importing the url from TAGS_ROUTES and performing the fetch request like we have here, it'd probably be best to use the getTags() function in the tags.service.ts file. That tags service has all the fetch request code abstracted in there so we can reuse it, see tags.component.ts line 55 as an example, it does the same thing passing a single text query.